File: problems/0003_longest-substring-without-repeating-characters/map/Solution.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        var map = new Dictionary<char, int>(256);
        int max = 0, start = 0, end = 0;
        foreach (char c in s) {
            int idx;
            if (!map.TryGetValue(c, out idx))
                idx = -1;
            map[c] = end++;
            if (idx == -1 || idx < start)
                max = Math.Max(max, end - start);
            else
                start = idx + 1;
        }
        return max;
    }
}

File: problems/0003_longest-substring-without-repeating-characters/map/solution.go
func lengthOfLongestSubstring(s string) int {
    m := make(map[byte]int, 256)
    max, l := 0, len(s)
    for start, end := 0, 0; end < l; {
        c := s[end]
        idx, ok := m[c]
        m[c] = end
        end += 1
        if !ok || idx < start {
            m := end - start
            if m > max {
                max = m
            }
        } else {
            start = idx + 1
        }
    }
    return max
}

